app/controllers/users_controller.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  before_action :set_user, only: %i[show edit update destroy]
+
+  # GET /users
+  # GET /users.json
+  def index
+    @users = User.all
+  end
+
+  # GET /users/1
+  # GET /users/1.json
+  def show
+    @upcoming_events = Event.upcoming
```

Comment:
- [ ] For user show page, we need to list the events created by the user, not all events.
- The user's show page displays: 
    - events the user has created separated in past and upcoming
    - events the user is attending separated in past and upcoming